src/hooks/query.ts

```diff
@@ -0,0 +1,116 @@
+import { useCallback, useEffect, useState } from 'react';
```

Comment:
I didn't trim it down to just the `useHashQuery` function since the others might be useful other places in the future. Do say if I should had trimmed it down instead. Also, I could had changed the usage to a `useHashQueryParameter` instead for the `LoginPage.tsx` now that I think about it.